### PR TITLE
feat: implement events for ctx

### PR DESCRIPTION
**Pull request by tymmesyde**
For modals and notifications

### CODE REVIEW

File: src/runtime/msg/action.rs [hunk 1]
```diff
@@ -60,6 +60,7 @@ pub enum ActionCtx {
     ///
     /// [`LibraryItem`]: crate::types::library::LibraryItem
     PullNotifications,
+    GetEvents,
```
Comment (elpiel) [inline src/runtime/msg/action.rs:64]: - [ ] Please, Document

File: src/runtime/msg/internal.rs [hunk 2]
```diff
@@ -118,4 +118,8 @@ pub enum Internal {
     SeekLogsResult(SeekLogRequest, Result<SuccessResponse, CtxError>),
     /// The result of querying the data for LocalSearch
     LoadLocalSearchResult(Url, Result<Vec<Searchable>, EnvError>),
+    /// Result for getModal request
+    GetModalResult(APIRequest, Result<GetModalResponse, CtxError>),
+    /// Result for getNotification request
+    GetNotificationResult(APIRequest, Result<GetNotificationResponse, CtxError>),
```
Comment (elpiel) [inline src/runtime/msg/internal.rs]: Can we have `null` responses when there is no Modal or Notification?

File: src/types/api/response.rs [hunk 3]
```diff
@@ -74,6 +74,33 @@ pub enum LinkDataResponse {
...
+    pub manifest_url: String,
+}
+
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct GetModalResponse {
+    pub id: String,
+    pub title: String,
+    pub message: String,
+    pub image_url: String,
```
Comment (elpiel) [inline src/types/api/response.rs]:
```suggestion
    pub image_url: url::Url,
```

File: src/types/api/response.rs [hunk 4]
```diff
@@ -74,6 +74,33 @@ pub enum LinkDataResponse {
     AuthKey(LinkAuthKey),
 }
 
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct ModalAddon {
+    pub name: String,
+    pub manifest_url: String,
```
Comment (elpiel) [inline src/types/api/response.rs]:
```suggestion
    pub manifest_url: url::Url,
```

File: src/types/api/response.rs [hunk 5]
```diff
@@ -74,6 +74,33 @@ pub enum LinkDataResponse {
...
+    pub external_url: Option<String>,
+}
+
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct GetNotificationResponse {
+    pub id: String,
+    pub title: String,
+    pub message: String,
+    pub external_url: Option<String>,
```
Comment (elpiel) [inline src/types/api/response.rs]:
```suggestion
    pub external_url: Option<url::Url>,
```

**Review by elpiel — COMMENTED** [review]
LGTM so far

File: src/runtime/msg/internal.rs (hunk 2 above)
Comment (tymmesyde) [reply src/runtime/msg/internal.rs]: It should not, the api respond with this pattern: 
```
{ result: ..., error: ... }
```

**Review by elpiel — APPROVED** [review]

**Pull request merged**